cmd/fts/cmd/run.go

```diff
@@ -0,0 +1,170 @@
+	runCmd.Flags().BoolVarP(&mainnet, "mainnet", "n", false, "Sets staging limits based on localnet or mainnet.")
+	runCmd.Flags().BoolVarP(&retryErrors, "retry", "e", false, "Retry tasks with error status")
+	runCmd.Flags().StringVar(&taskFolder, "folder", "", "Folder with organized tasks of directories or files")
+	runCmd.Flags().BoolVarP(&hiddenFiles, "all", "a", false, "Include hidden files & folders from top level folder")
+	runCmd.Flags().StringVar(&ipfsrevproxy, "ipfsrevproxy", "127.0.0.1:6002", "Powergate IPFS reverse proxy multiaddr")
+	runCmd.Flags().StringVarP(&resultsOut, "results", "r", "results.json", "The location to store intermediate and final results.")
+
+	// Not included in the public commands
+	runCmd.Flags().Int64Var(&concurrent, "concurrent", 10000, "Max concurrent tasks being processed")
+	runCmd.Flags().Int64Var(&maxStagedBytes, "max-staged-bytes", 0, "Maximum bytes of all tasks queued on staging")
+	runCmd.Flags().Int64Var(&maxDealBytes, "max-deal-bytes", 0, "Maximum bytes of a single deal")
+	runCmd.Flags().Int64Var(&minDealBytes, "min-deal-bytes", 0, "Minimum bytes of a single deal")
+	runCmd.Flags().MarkHidden("concurrent")
+	runCmd.Flags().MarkHidden("max-staged-bytes")
+	runCmd.Flags().MarkHidden("max-deal-bytes")
+	runCmd.Flags().MarkHidden("min-deal-bytes")
+
+	rootCmd.AddCommand(runCmd)
+}
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Run a storage task pipeline",
+	Long:  `Run a storage task pipeline to migrate large collections of data to Filecoin`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := viper.BindPFlags(cmd.Flags())
+		checkErr(err)
+		// set default variables
+		if mainnet {
+			if maxStagedBytes == 0 {
+				maxStagedBytes = 26843545600 // 25Gib
+			}
+			if maxDealBytes == 0 {
+				maxDealBytes = 4294967296 // 4Gib
+			}
+			if minDealBytes == 0 {
+				minDealBytes = 67108864 // 0.5Gib
+			}
+		} else { // localnet
+			if maxStagedBytes == 0 {
+				maxStagedBytes = 524288
+			}
+			if maxDealBytes == 0 {
+				maxDealBytes = 524288
+			}
+			if minDealBytes == 0 {
+				minDealBytes = 9000
+			}
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if maxStagedBytes < maxDealBytes {
+			Fatal(fmt.Errorf("Max deal size (%d) is larger than max staging size (%d)", maxDealBytes, maxStagedBytes))
+		}
+
+		// Read all tasks from the input folder
+		unfiltered, err := pathToTasks(taskFolder)
+		checkErr(err)
+		Message("Input tasks: %d", len(unfiltered))
+		allTasks := filterTasks(unfiltered)
+
+		// Ensure there aren't existing tasks in the same queue
+		knownTasks, err := openResults(resultsOut)
+		checkErr(err)
+		Message("Existing tasks: %d", len(knownTasks))
+		allTasks = mergeNewTasks(knownTasks, allTasks)
+
+		pendingTasks := cleanErrors(removeComplete(allTasks), retryErrors)
+
+		Message("Pending tasks: %d", len(pendingTasks))
+		time.Sleep(1 * time.Second)
```

Comment:
added a little sleep here otherwise the summary up to now can happen too fast to be seen at all